### PR TITLE
[PPML] TrustedML for MLlib and LGBM

### PR DESCRIPTION
**Pull request by Uxito-Ada**
## Description

Trusted machine learning image containing secure Spark and LGBM toolkit

### 3. Summary of the change 

as title

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

LGBM in our private repo
libssl-dev (openssl)

### CODE REVIEW

**Comment by qiyuangong** [comment]
This PR includes too many file changes. Can we split it into several PRs?

**Comment by Uxito-Ada** [comment]
This PR mainly includes scripts for building a Spark MLlib image, and a basic setup for LGBM.
Detailed codes and patches of LGBM are in our private repo, and is cloned into the image.

We will going to merge this fundamental PR after @SEULiyao refines the Spark part, and continue to append K8S distributed LGBM in the other PR later.

@qiyuangong

File: ppml/trusted-machine-learning/make-lgbm.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (qiyuangong) [inline ppml/trusted-machine-learning/make-lgbm.sh:1]: Remove LGTM build from this PR. Replace it with download from FPT.

File: ppml/trusted-machine-learning/spark-driver-template.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        path: /var/run/aesmd/aesm.socket
+    - name: secure-keys
+      secret:
+        secretName: ssl-keys
+    - name: nfs-storage
+      persistentVolumeClaim:
+        claimName: nfsvolumeclaim
+    - name: kubeconfig
+      secret:
+        secretName: kubeconfig-secret
```
Comment (qiyuangong) [inline ppml/trusted-machine-learning/spark-driver-template.yaml:86]: Add an empty line.

File: ppml/trusted-machine-learning/spark-executor-template.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+apiVersion: v1
```
Comment (qiyuangong) [inline ppml/trusted-machine-learning/spark-executor-template.yaml:1]: If we have duplicate yaml, we need to merge them.

**Comment by qiyuangong** [comment]
> This PR mainly includes scripts for building a Spark MLlib image, and a basic setup for LGBM. Detailed codes and patches of LGBM are in our private repo, and is cloned into the image.
> 
> We will going to merge this fundamental PR after @SEULiyao refines the Spark part, and continue to append K8S distributed LGBM in the other PR later.
> 
> @qiyuangong

Need to remove duplicate files, docs and readme in this PR. Meanwhile, we can move building LGTM to offline and only provide lib download in dockerfile.

File: ppml/trusted-machine-learning/spark-executor-template.yaml (hunk 3 above)
Comment (Uxito-Ada) [reply ppml/trusted-machine-learning/spark-executor-template.yaml]: Yamls for driver and executor are different, for example the driver requires a kubeconfig file to create pod on k8s while the executor does not.

**Comment by Uxito-Ada** [comment]
Going to add k8s LGBM, k8s MLlib and LGBM document, separate MLlib section from README and download LGBM targets from FTP in the other PR.

**Pull request merged**